src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/exceptionhandlers/TaskErrorHandler.java

```diff
@@ -71,4 +76,10 @@ private void registerEvent(EventRelatedThrowable exception, String reason) {
             processEvent.getEvent()
         );
     }
+
+    private void incrementUploadFailureCount(DocUploadFailureGenericException exc) {
```

Comment:
My guess would be this method is entirely envelope related - nothing additionally will be needed for exception handling therefore it can accept `Envelope` straight away. Current set of methods accepts exception because of generalisation

Just a thought

Reply:
Taking exception as param here for consistency as well. That's 3rd `if` already. We may want to use a map from exception to function that handles it soon (before PMD starts complaining ;))